src/tests/krate.rs

```diff
@@ -1589,6 +1589,120 @@ fn publish_after_yank_max_version() {
     assert_eq!(json.krate.max_version, "2.0.0");
 }
 
+#[test]
+fn publish_records_an_audit_action() {
+    use cargo_registry::models::{VersionAction, VersionOwnerAction};
+
+    let (app, anon, _, token) = TestApp::full().with_token();
+
+    app.db(|conn| assert!(VersionOwnerAction::all(&conn).unwrap().is_empty()));
+
+    // Upload a new crate, putting it in the git index
+    let crate_to_publish = PublishBuilder::new("fyk");
+    token.enqueue_publish(crate_to_publish).good();
+
+    // make sure it doesn't have any audit actions
+    // do this as a full integration test once the api is in place.
+    let json = anon.show_version("fyk", "1.0.0");
+    let version_id = json.version.id;
+
+    // make sure it has one audit action
+    // do this as a full integration test once the api is in place.
+    app.db(|conn| {
+        let actions = VersionOwnerAction::all(conn).unwrap();
+        assert_eq!(actions.len(), 1);
+        let action = actions[0];
+        assert_eq!(action.version_id, version_id);
+        assert_eq!(action.user_id, token.as_model().user_id);
+        assert_eq!(action.api_token_id, Some(token.as_model().id));
+        assert_eq!(action.action, VersionAction::Publish);
+    });
+}
+
+#[test]
+fn yank_records_an_audit_action() {
+    use cargo_registry::models::{VersionAction, VersionOwnerAction};
+
+    let (app, anon, _, token) = TestApp::full().with_token();
+
+    // Upload a new crate, putting it in the git index
+    let crate_to_publish = PublishBuilder::new("fyk");
+    token.enqueue_publish(crate_to_publish).good();
+
+    // make sure it doesn't have any audit actions
+    // do this as a full integration test once the api is in place.
+    let json = anon.show_version("fyk", "1.0.0");
+    let version_id = json.version.id;
+    app.db(|conn| {
+        assert!(
+            VersionOwnerAction::by_version_id_and_action(conn, version_id, VersionAction::Yank)
+                .unwrap()
+                .is_empty()
+        )
+    });
+
+    // yank it
+    token.yank("fyk", "1.0.0").good();
+
+    // make sure it has one audit action
```

Comment:
...and at this point, there will be 2 total audit actions, but one `Yank` audit action, so I think this comment should be updated to say "make sure it has one yank audit action" or "make sure it has one publish and one yank audit action" (since the API will probably return all audit actions rather than have a way to do filtering as we can do with the database here)